users/views.py

```diff
@@ -65,6 +46,14 @@ def create(self, request, *args, **kwargs):
     """
 
     def partial_update(self, request, *args, **kwargs):
+        # Return 401 if no token found
+        if request.auth is None:
+            return Response(status=HTTP_401_UNAUTHORIZED)
+
+        # Return 401 if token doesn't match with the lookup value
+        if request.user.username != kwargs[self.lookup_field]:
+            return Response(status=HTTP_401_UNAUTHORIZED)
```

Comment:
401 is for when a token isn't included in the request. Since a token mismatch means someone is trying to modify someone else's profile, this should return a 403 (Forbidden)